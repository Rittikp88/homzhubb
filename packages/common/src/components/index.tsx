// Atoms
export { Text, Label, TextSizeType, FontWeightType, TextFieldType } from '@homzhub/common/src/components/atoms/Text';
export { Button, ButtonType } from '@homzhub/common/src/components/atoms/Button';
export { ImageRound, ImageSquare, Image } from '@homzhub/common/src/components/atoms/Image';
export { OtpTimer } from '@homzhub/common/src/components/atoms/OtpTimer';

// Molecules
export { FormTextInput } from '@homzhub/common/src/components/molecules/FormTextInput';
export { FormButton } from '@homzhub/common/src/components/molecules/FormButton';
export { Header } from '@homzhub/common/src/components/molecules/Header';
export { OtpInputs } from '@homzhub/common/src/components/molecules/OtpInputs';
export { TermsCondition } from '@homzhub/common/src/components/molecules/TermsAndCondition';
export { WithFieldError } from '@homzhub/common/src/components/molecules/WithFieldError';

// Organism
export { SignUpForm } from '@homzhub/common/src/components/organisms/SignUpForm';

// Templates
export { SignupView } from '@homzhub/common/src/components/templates/SignupView';
