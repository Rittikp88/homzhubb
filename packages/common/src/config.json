{
  "API_BASE_URL": "https://jsonplaceholder.typicode.com",
  "OTP_LENGTH": 6
}
