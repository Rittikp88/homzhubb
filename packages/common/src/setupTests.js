/* eslint-disable */
// @ts-nocheck
require('../../../setupTests');

jest.mock('@ptomasroos/react-native-multi-slider', () => {});
