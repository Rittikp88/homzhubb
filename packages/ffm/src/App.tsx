import React, { useEffect, useState } from 'react';
import { Provider } from 'react-redux';
import FlashMessage, { MessageComponentProps } from 'react-native-flash-message';
import { configureStore } from '@homzhub/common/src/modules/store';
import { I18nService } from '@homzhub/common/src/services/Localization/i18nextService';
import { AnalyticsService } from '@homzhub/common/src/services/Analytics/AnalyticsService';
import { StoreProviderService } from '@homzhub/common/src/services/StoreProviderService';
import { StorageKeys, StorageService } from '@homzhub/common/src/services/storage/StorageService';
import { CommonActions } from '@homzhub/common/src/modules/common/actions';
import ErrorBoundary from '@homzhub/mobile/src/components/HOC/ErrorBoundary';
import { Toast } from '@homzhub/common/src/components/molecules/Toast';
import { RootNavigator } from '@homzhub/ffm/src/navigation/RootNavigator';
import { SupportedLanguages } from '@homzhub/common/src/services/Localization/constants';

StoreProviderService.init(configureStore);
const store = StoreProviderService.getStore();

const App: () => React.ReactElement = () => {
  const [booting, setBooting] = useState(true);
  useEffect(() => {
    bootUp().then();
  }, []);

  const bootUp = async (): Promise<void> => {
    store.dispatch(CommonActions.getCountries());
    const selectedLanguage: SupportedLanguages | null = await StorageService.get(StorageKeys.USER_SELECTED_LANGUAGE);
    await I18nService.init(selectedLanguage || undefined);
    await AnalyticsService.initMixpanel();

    setBooting(false);
  };

  const renderToast = (props: MessageComponentProps): React.ReactElement => <Toast {...props} />;

  return (
    <ErrorBoundary>
      <Provider store={store}>
        <RootNavigator booting={booting} />
        <FlashMessage position="bottom" MessageComponent={renderToast} />
      </Provider>
    </ErrorBoundary>
  );
};

export default App;
