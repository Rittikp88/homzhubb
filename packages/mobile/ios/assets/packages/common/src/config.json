{
  "API_BASE_URL": "http://3.7.27.67/api/v1/",
  "OTP_LENGTH": 6
}
