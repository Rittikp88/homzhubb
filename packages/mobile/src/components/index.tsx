// Atoms
export { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
export { Loader } from '@homzhub/mobile/src/components/atoms/Loader';
export { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
export { StatusBarComponent } from '@homzhub/mobile/src/components/atoms/StatusBar';
export { RNVideo } from '@homzhub/mobile/src/components/atoms/Video';

// Molecules
export { AssetDetailsImageCarousel } from '@homzhub/mobile/src/components/molecules/AssetDetailsImageCarousel';
export { AssetRatings } from '@homzhub/mobile/src/components/molecules/AssetRatings';
export { BottomSheet } from '@homzhub/mobile/src/components/molecules/BottomSheet';
export { BottomSheetListView } from '@homzhub/mobile/src/components/molecules/BottomSheetListView';
export { ButtonGroup } from '@homzhub/mobile/src/components/molecules/ButtonGroup';
export { CardBody } from '@homzhub/mobile/src/components/molecules/CardBody';
export { CollapsibleSection } from '@homzhub/mobile/src/components/molecules/CollapsibleSection';
export { CurrentLocation } from '@homzhub/mobile/src/components/molecules/CurrentLocation';
export { FullScreenAssetDetailsCarousel } from '@homzhub/mobile/src/components/molecules/FullScreenAssetDetailsCarousel';
export { GradientBackground } from '@homzhub/mobile/src/components/molecules/GradientBackground';
export { Header } from '@homzhub/mobile/src/components/molecules/Header';
export { LeaseDetailsForm } from '@homzhub/mobile/src/components/molecules/LeaseDetailsForm';
export { MaintenanceDetails } from '@homzhub/mobile/src/components/molecules/MaintenanceDetails';
export { OtpInputs } from '@homzhub/mobile/src/components/molecules/OtpInputs';
export { PropertyAmenities } from '@homzhub/mobile/src/components/molecules/PropertyAmenities';
export { PropertyDetailsLocation } from '@homzhub/mobile/src/components/molecules/PropertyDetailsLocation';
export { PropertyListImageCarousel } from '@homzhub/mobile/src/components/molecules/PropertyListImageCarousel';
export { PropertyMapCard } from '@homzhub/mobile/src/components/molecules/PropertyMapCard';
export { Range } from '@homzhub/mobile/src/components/molecules/Range';
export { ResaleDetailsForm } from '@homzhub/mobile/src/components/molecules/ResaleDetailsForm';
export { RoomsFilter } from '@homzhub/mobile/src/components/molecules/RoomsFilter';
export { SearchBar } from '@homzhub/mobile/src/components/molecules/SearchBar';
export { SearchResults } from '@homzhub/mobile/src/components/molecules/SearchResults';
export { SetLocationForm } from '@homzhub/mobile/src/components/molecules/SetLocationForm';
export { ShieldGroup } from '@homzhub/mobile/src/components/molecules/ShieldGroup';
export { StepIndicatorComponent } from '@homzhub/mobile/src/components/molecules/StepIndicator';
export { TimeSlotGroup } from '@homzhub/mobile/src/components/molecules/TimeSlotGroup';
export { Toast } from '@homzhub/mobile/src/components/molecules/Toast';

// Organisms
export { AssetTypeFilter } from '@homzhub/mobile/src/components/organisms/AssetTypeFilter';
export { CheckoutAssetDetails } from '@homzhub/mobile/src/components/organisms/CheckoutAssetDetails';
export { PaymentSuccess } from '@homzhub/mobile/src/components/organisms/PaymentSuccess';
export { PropertyDetailsItems } from '@homzhub/mobile/src/components/organisms/PropertyDetailItems';
export { PropertyPayment } from '@homzhub/mobile/src/components/organisms/PropertyPayment';
// TODO: More files to be added

// Templates
export { AnimatedServiceList } from '@homzhub/mobile/src/components/templates/AnimatedServiceList';
