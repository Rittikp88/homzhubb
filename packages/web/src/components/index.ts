// atoms
export { ProgressBar } from '@homzhub/web/src/components/atoms/ProgressBar';
export { SearchField } from '@homzhub/web/src/components/atoms/SearchField';

// molecules
export { PendingPropertiesCard } from '@homzhub/web/src/components/molecules/PendingPropertiesCard';
export { NextPrevBtn } from '@homzhub/web/src/components/molecules/NextPrevBtn';

// organisms
