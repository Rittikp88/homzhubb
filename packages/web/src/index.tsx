import { AppRegistry } from 'react-native';

import { App } from './App';

AppRegistry.registerComponent('homzhub', () => App);
AppRegistry.runApplication('homzhub', {
  rootTag: document.getElementById('root'),
});
