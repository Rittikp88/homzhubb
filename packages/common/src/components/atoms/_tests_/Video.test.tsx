import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { RNVideo } from '@homzhub/common/src/components';

describe('Badge', () => {
  const props = {
    uri: 'videoUrl',
    onBuffer: jest.fn(),
    onVideoError: jest.fn(),
  };
  const wrapper: ShallowWrapper = shallow(<RNVideo {...props} />);

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
