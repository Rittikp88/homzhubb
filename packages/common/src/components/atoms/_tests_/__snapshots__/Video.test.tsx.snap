// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Video
    onEnd={[Function]}
    onFullScreen={false}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    paused={false}
    resizeMode="content"
    source={
      Object {
        "uri": "videoUrl",
      }
    }
    style={
      Object {
        "backgroundColor": "#000",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    volume={1}
  />
  <MediaControls
    duration={0}
    isFullScreen={false}
    isLoading={true}
    mainColor="#000"
    onFullScreen={[Function]}
    onPaused={[Function]}
    onReplay={[Function]}
    onSeek={[Function]}
    onSeeking={[Function]}
    playerState={0}
    progress={0}
  />
</View>
`;
